graph-proxy/src/graphql/workflows.rs

```diff
@@ -309,6 +312,295 @@ impl TryFrom<TaskMap> for Vec<Task> {
     }
 }
 
+/// A Workflow consisting of one or more [`Task`]s
+#[derive(Debug, SimpleObject)]
+struct LazyWorkflow {
+    /// The name given to the workflow, unique within a given visit
+    name: String,
+    /// The time at which the workflow began running
+    status: LazyWorkflowStatus,
+}
+
+/// Metadata of a workflow
+#[derive(Debug)]
+struct WorkflowMetaData {
+    /// Workflow name
+    workflow_name: String,
+    /// Namespace in which workflow exists
+    namespace: String,
+}
+
+impl TryFrom<IoArgoprojWorkflowV1alpha1Workflow> for LazyWorkflow {
+    type Error = WorkflowParsingError;
+
+    fn try_from(value: IoArgoprojWorkflowV1alpha1Workflow) -> Result<Self, Self::Error> {
+        let metadata = WorkflowMetaData {
+            workflow_name: value.metadata.name.clone().unwrap(),
+            namespace: value.metadata.namespace.unwrap(),
+        };
+        Ok(Self {
+            name: value.metadata.name.unwrap(),
+            status: LazyWorkflowStatus::try_from((value.status.unwrap(), metadata))?,
+        })
+    }
+}
+lazy_static! {
+    static ref CLIENT: reqwest::Client = reqwest::Client::new();
+}
+#[allow(clippy::missing_docs_in_private_items)]
+macro_rules! fetch_tasks {
+    ($namespace:expr, $workflow_name:expr, $ctx:expr) => {{
```

Comment:
Think we could have this macro also create the function definition - just to compact things a little more